### PR TITLE
Adding negative function to maps

### LINKED ISSUE #1205: maps handling negative values better
### Describe the bug

Originally, OED did not have negative readings but it is now allowed. If you place a negative value on a map, you see no circle. If you place the cursor over the expected location you see the correct hover. This screenshot shows this:

image

The issue is likely that src/client/app/containers/MapChartContainer.ts finds the max size where it should probably use the max - min size or something like this.

Another option might be to use another visual for negative (lines through color circle or something else) so it is clear to the user that some values are negative. This would be combined with fixing the actual size.

If others have thoughts about the look then please state them. The ideas here would have larger negative values have smaller circles.

### To Reproduce

Steps to reproduce the behavior (assuming you have the usual developer testData loaded):
1. Create a new meter where the important values are:

image

2. Place these values in a CSV file for the readings:

reading,start time,end time
-10,2023-01-01 00:00:00,2024-01-01 00:00:00

and do a CSV Readings upload where you specify the meter (from step 1), header row and refresh the daily/hourly readings, along with the CSV created in this step.
3. Go to the map and graph this meter.
4. Place cursor over Library to see

Note the provided CSV reading value is a constant -10 across all of 2023. It could be modified to get other values. If you want to place a circle in another location, looking at the location/GPS value for the amp meters will help.

### Expected behavior

The circle size should be visible and scale with usage.

### Additional context

None

### PR DESCRIPTION
**Pull request by mmehta2669**
# Description

Added changes to maps to handle negative values

Fixes #1205 

## Type of change

(Check the ones that apply by placing an "x" instead of the space in the [ ] so it becomes [x])

- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

(Note what you have done by placing an "x" instead of the space in the [ ] so it becomes [x]. It is hoped you do all of them.)

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.github.io/developer/cla.html) and each author is listed in the Description section.

## Limitations
N/A

### CODE REVIEW

**Comment by mmehta2669** [comment]
Thanks @huss for the suggestions on the code. All the suggestions were really helpful to understand how Plotly works.

**Comment by Rakesh-Ranga-Buram** [comment]
Everything looks fine to me. Please convert this draft to PR

File: src/client/app/components/MapChartComponent.tsx [hunk 1]
```diff
@@ -289,22 +290,105 @@ export default function MapChartComponent() {
 					}
 				}
 			}
+
+			// TODO DEBUG Using amp 1, 2 & 3 from test data that within map. This arbitrarily changes the value for testing.
```
Comment (huss) [inline src/client/app/components/MapChartComponent.tsx]: Debugging statements that are commented out should be removed before the code is added to OED.

**Review by huss — CHANGES_REQUESTED** [review]
Thanks to @mmehta2669 for another contribution. Review and testing found it works as desired. I have made one small comment to address.

**Review by huss — APPROVED** [review]
Thanks to @mmehta2669 for addressing my comment. Everything looks good. Congratulations on another accepted contribution.

**Pull request merged**